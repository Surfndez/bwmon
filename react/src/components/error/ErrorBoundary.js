import React, { Component } from 'react';
import Error from '../messages/Error';

export default class ErrorBoundary extends Component {
    state = { error: null, errorInfo: null };

    componentDidCatch(error, errorInfo) {
        this.setState({
            error: error,
            errorInfo: errorInfo,
        });
    }

    renderError = () => (this.state.error ? <Error message={this.state.error.toString()} /> : null);
    renderErrorInfo = () => (this.state.errorInfo ? <div>{this.state.errorInfo.componentStack}</div> : null);

    render() {
        if (this.state.error || this.state.errorInfo) {
            return (
                <div>
                    {this.renderError()}
                    {this.renderErrorInfo()}
                </div>
            );
        }
        return this.props.children;
    }
}
