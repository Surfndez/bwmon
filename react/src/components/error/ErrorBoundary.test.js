import React from 'react';
import { create } from 'react-test-renderer';
import ErrorBoundary from './ErrorBoundary';

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const Something = () => null;

describe('<ErrorBoundary> window', () => {
    test('renders correctly', () => {
        const tree = create(
            <ErrorBoundary>
                <Something />
            </ErrorBoundary>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('renders error', () => {
        const error = new Error('Error!!!');
        const wrapper = mount(
            <ErrorBoundary>
                <Something />
            </ErrorBoundary>
        );
        expect(wrapper).toMatchSnapshot();
        wrapper.find(Something).simulateError(error);
        expect(wrapper).toMatchSnapshot();
    });
});
