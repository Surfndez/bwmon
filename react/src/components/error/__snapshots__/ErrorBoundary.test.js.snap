// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary> window renders correctly 1`] = `null`;

exports[`<ErrorBoundary> window renders error 1`] = `
<ErrorBoundary>
  <Something />
</ErrorBoundary>
`;

exports[`<ErrorBoundary> window renders error 2`] = `
<ErrorBoundary>
  <div>
    <Error
      message="Error: Error!!!"
    >
      <Message
        message="Error: Error!!!"
        severity="error"
      >
        <div
          className="makeStyles-root-1"
        >
          <WithStyles(ForwardRef(Alert))
            severity="error"
          >
            <ForwardRef(Alert)
              classes={
                Object {
                  "action": "MuiAlert-action",
                  "filledError": "MuiAlert-filledError",
                  "filledInfo": "MuiAlert-filledInfo",
                  "filledSuccess": "MuiAlert-filledSuccess",
                  "filledWarning": "MuiAlert-filledWarning",
                  "icon": "MuiAlert-icon",
                  "message": "MuiAlert-message",
                  "outlinedError": "MuiAlert-outlinedError",
                  "outlinedInfo": "MuiAlert-outlinedInfo",
                  "outlinedSuccess": "MuiAlert-outlinedSuccess",
                  "outlinedWarning": "MuiAlert-outlinedWarning",
                  "root": "MuiAlert-root",
                  "standardError": "MuiAlert-standardError",
                  "standardInfo": "MuiAlert-standardInfo",
                  "standardSuccess": "MuiAlert-standardSuccess",
                  "standardWarning": "MuiAlert-standardWarning",
                }
              }
              severity="error"
            >
              <WithStyles(ForwardRef(Paper))
                className="MuiAlert-root MuiAlert-standardError"
                elevation={0}
                role="alert"
                square={true}
              >
                <ForwardRef(Paper)
                  className="MuiAlert-root MuiAlert-standardError"
                  classes={
                    Object {
                      "elevation0": "MuiPaper-elevation0",
                      "elevation1": "MuiPaper-elevation1",
                      "elevation10": "MuiPaper-elevation10",
                      "elevation11": "MuiPaper-elevation11",
                      "elevation12": "MuiPaper-elevation12",
                      "elevation13": "MuiPaper-elevation13",
                      "elevation14": "MuiPaper-elevation14",
                      "elevation15": "MuiPaper-elevation15",
                      "elevation16": "MuiPaper-elevation16",
                      "elevation17": "MuiPaper-elevation17",
                      "elevation18": "MuiPaper-elevation18",
                      "elevation19": "MuiPaper-elevation19",
                      "elevation2": "MuiPaper-elevation2",
                      "elevation20": "MuiPaper-elevation20",
                      "elevation21": "MuiPaper-elevation21",
                      "elevation22": "MuiPaper-elevation22",
                      "elevation23": "MuiPaper-elevation23",
                      "elevation24": "MuiPaper-elevation24",
                      "elevation3": "MuiPaper-elevation3",
                      "elevation4": "MuiPaper-elevation4",
                      "elevation5": "MuiPaper-elevation5",
                      "elevation6": "MuiPaper-elevation6",
                      "elevation7": "MuiPaper-elevation7",
                      "elevation8": "MuiPaper-elevation8",
                      "elevation9": "MuiPaper-elevation9",
                      "outlined": "MuiPaper-outlined",
                      "root": "MuiPaper-root",
                      "rounded": "MuiPaper-rounded",
                    }
                  }
                  elevation={0}
                  role="alert"
                  square={true}
                >
                  <div
                    className="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
                    role="alert"
                  >
                    <div
                      className="MuiAlert-icon"
                    >
                      <ForwardRef(ErrorOutlineIcon)
                        fontSize="inherit"
                      >
                        <WithStyles(ForwardRef(SvgIcon))
                          fontSize="inherit"
                        >
                          <ForwardRef(SvgIcon)
                            classes={
                              Object {
                                "colorAction": "MuiSvgIcon-colorAction",
                                "colorDisabled": "MuiSvgIcon-colorDisabled",
                                "colorError": "MuiSvgIcon-colorError",
                                "colorPrimary": "MuiSvgIcon-colorPrimary",
                                "colorSecondary": "MuiSvgIcon-colorSecondary",
                                "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                "root": "MuiSvgIcon-root",
                              }
                            }
                            fontSize="inherit"
                          >
                            <svg
                              aria-hidden={true}
                              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                              />
                            </svg>
                          </ForwardRef(SvgIcon)>
                        </WithStyles(ForwardRef(SvgIcon))>
                      </ForwardRef(ErrorOutlineIcon)>
                    </div>
                    <div
                      className="MuiAlert-message"
                    >
                      <WithStyles(ForwardRef(AlertTitle))>
                        <ForwardRef(AlertTitle)
                          classes={
                            Object {
                              "root": "MuiAlertTitle-root",
                            }
                          }
                        >
                          <WithStyles(ForwardRef(Typography))
                            className="MuiAlertTitle-root"
                            component="div"
                            gutterBottom={true}
                          >
                            <ForwardRef(Typography)
                              className="MuiAlertTitle-root"
                              classes={
                                Object {
                                  "alignCenter": "MuiTypography-alignCenter",
                                  "alignJustify": "MuiTypography-alignJustify",
                                  "alignLeft": "MuiTypography-alignLeft",
                                  "alignRight": "MuiTypography-alignRight",
                                  "body1": "MuiTypography-body1",
                                  "body2": "MuiTypography-body2",
                                  "button": "MuiTypography-button",
                                  "caption": "MuiTypography-caption",
                                  "colorError": "MuiTypography-colorError",
                                  "colorInherit": "MuiTypography-colorInherit",
                                  "colorPrimary": "MuiTypography-colorPrimary",
                                  "colorSecondary": "MuiTypography-colorSecondary",
                                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                  "displayBlock": "MuiTypography-displayBlock",
                                  "displayInline": "MuiTypography-displayInline",
                                  "gutterBottom": "MuiTypography-gutterBottom",
                                  "h1": "MuiTypography-h1",
                                  "h2": "MuiTypography-h2",
                                  "h3": "MuiTypography-h3",
                                  "h4": "MuiTypography-h4",
                                  "h5": "MuiTypography-h5",
                                  "h6": "MuiTypography-h6",
                                  "noWrap": "MuiTypography-noWrap",
                                  "overline": "MuiTypography-overline",
                                  "paragraph": "MuiTypography-paragraph",
                                  "root": "MuiTypography-root",
                                  "srOnly": "MuiTypography-srOnly",
                                  "subtitle1": "MuiTypography-subtitle1",
                                  "subtitle2": "MuiTypography-subtitle2",
                                }
                              }
                              component="div"
                              gutterBottom={true}
                            >
                              <div
                                className="MuiTypography-root MuiAlertTitle-root MuiTypography-body1 MuiTypography-gutterBottom"
                              >
                                Error: Error!!!
                              </div>
                            </ForwardRef(Typography)>
                          </WithStyles(ForwardRef(Typography))>
                        </ForwardRef(AlertTitle)>
                      </WithStyles(ForwardRef(AlertTitle))>
                    </div>
                  </div>
                </ForwardRef(Paper)>
              </WithStyles(ForwardRef(Paper))>
            </ForwardRef(Alert)>
          </WithStyles(ForwardRef(Alert))>
        </div>
      </Message>
    </Error>
    <div>
      
    in Something (created by ErrorBoundary)
    in ErrorBoundary (created by WrapperComponent)
    in WrapperComponent
    </div>
  </div>
</ErrorBoundary>
`;
